code/modules/economy/economy_admin.dm

```diff
@@ -0,0 +1,91 @@
+
+/datum/ui_module/economy_manager
+	name = "Economy Manager"
+
+/datum/ui_module/economy_manager/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.admin_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "EconomyManager", name, 600, 350, master_ui, state)
+		ui.autoupdate = TRUE
+		ui.open()
+
+/datum/ui_module/economy_manager/ui_data(mob/user)
+	var/list/data = list()
+
+	data["next_payroll_time"] = round((SSeconomy.next_paycheck_delay - world.time) / (60 SECONDS), 0.1)
+
+	return data
+
+/datum/ui_module/economy_manager/ui_act(action, params, datum/tgui/ui)
+	if(..())
+		return
+
+	switch(action)
+		if("payroll_modification")
+			var/list/accounts_to_modify = list()
+			var/num_input = input(ui.user, "Amount (If for more than 4-5 people, Keep this between +/-$100 for the love of god)", "How much to modify by?") as null|num
+			if(!num_input)
+				return
+			switch(params["mod_type"])
+				if("global")
+					if(num_input > 0)
+						SSeconomy.global_paycheck_bonus += num_input
+					else
+						SSeconomy.global_paycheck_deduction += -num_input
+					log_and_message_admins("has modified the global payroll by [num_input].")
+				if("department")
+					var/department_input = input(ui.user, "Department", "Which Department Account?") in (GLOB.station_departments - list(DEPARTMENT_ASSISTANT, DEPARTMENT_SILICON))
+					if(!department_input)
+						return
+					var/datum/money_account/department_account = GLOB.station_money_database.get_account_by_department(department_input)
+					if(!department_account)
+						return
+					accounts_to_modify += department_account
+					log_and_message_admins("has modified the payroll of the [department_input] department by [num_input].")
+				if("department_members")
+					var/department_input = input(ui.user, "Department", "Members from Which Department?") in (GLOB.station_departments - list(DEPARTMENT_ASSISTANT, DEPARTMENT_SILICON))
+					var/datum/station_department/department = get_department_from_name(department_input)
+					if(!department)
+						return
+					for(var/datum/department_member/member as anything in department.members)
+						if(member.member_account)
+							accounts_to_modify += member.member_account
+					log_and_message_admins("has modified the payroll of the [department_input] department's members by [num_input].")
+				if("crew_member")
+					var/account_input = input(ui.user, "Account Number", "Please Specify Account Number.") as null|num
+					var/datum/money_account/account = GLOB.station_money_database.find_user_account(account_input, FALSE)
+					if(!account)
+						return
+					accounts_to_modify += account
+			if(!length(accounts_to_modify))
+				return
+			var/raw_input = input(ui.user, "Add message (Max 50 Characters):", "If you wish the owner(s) of the account(s) to recieve a message, please specify") as null|message
```

Comment:
Prob want to get the input before you log stuff of it happening

Reply:
fixed in next commit